pkg/apis/networking/validation/validation_test.go

```diff
@@ -1566,267 +1566,243 @@ func TestValidateIngressUpdate(t *testing.T) {
 	}
 }
 
+type netIngressTweak func(ingressClass *networking.IngressClass)
+
+func makeValidIngressClass() *networking.IngressClass {
+	return &networking.IngressClass{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test123",
+		},
+		Spec: networking.IngressClassSpec{
+			Controller: "foo.co/bar",
+		},
+	}
+}
+
+func makeValidIngressClassUpdateCustom(tweaks ...netIngressTweak) *networking.IngressClass {
```

Comment:
Sing a variadic arg can be empty, there's not much difference between these.  I'd suggest keep one -  the short name and the variadic arg.

Reply:
I'd also take name as an argument and maybe the controller string, too.

Reply:
See pkg/API/service/testing for example

Reply:
@thockin Done